lib/MetaCPAN/Script/Backpan.pm

```diff
@@ -3,66 +3,228 @@ package MetaCPAN::Script::Backpan;
 use strict;
 use warnings;
 
-use BackPAN::Index;
 use Moose;
 
+use Log::Contextual qw( :log :dlog );
+use BackPAN::Index;
+use MetaCPAN::Types qw( Bool HashRef Str );
+
 with 'MetaCPAN::Role::Script', 'MooseX::Getopt::Dashes';
 
+has distribution => (
+    is            => 'ro',
+    isa           => Str,
+    documentation => 'work on given distribution',
+);
+
+has undo => (
+    is            => 'ro',
+    isa           => Bool,
+    default       => 0,
+    documentation => 'mark releases as status=cpan',
+);
+
+has files_only => (
+    is            => 'ro',
+    isa           => Bool,
+    default       => 0,
+    documentation => 'only update the "file" index',
+);
+
+has _cpan_files_list => (
+    is      => 'ro',
+    isa     => HashRef,
+    lazy    => 1,
+    builder => '_build_cpan_files_list',
+);
+
+has _release_status => (
+    is      => 'ro',
+    isa     => HashRef,
+    default => sub { +{} },
+);
+
+has _bulk => (
+    is      => 'ro',
+    isa     => HashRef,
+    default => sub { +{} },
+);
+
+sub _build_cpan_files_list {
+    my $self = shift;
+    my $ls   = $self->cpan->file(qw(indices find-ls.gz));
+    unless ( -e $ls ) {
+        log_error {"File $ls does not exist"};
+        exit;
+    }
+    log_info {"Reading $ls"};
+    my $cpan = {};
+    open my $fh, "<:gzip", $ls;
+    while (<$fh>) {
+        my $path = ( split(/\s+/) )[-1];
+        next unless ( $path =~ /^authors\/id\/\w+\/\w+\/(\w+)\/(.*)$/ );
```

Comment:
Would be nice to use a different regex delimiter here so we don't need to escape the slashes, eg `m{...}`